src/main/java/io/reactorsolutions/actors/verticles/EnemyVerticle.java

```diff
@@ -1,59 +1,64 @@
-package io.reactorsolutions.vertx_kafka.verticles;
+package io.reactorsolutions.actors.verticles;
 
 import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class EnemyVerticle extends AbstractVerticle {
 
+  public static final String ENEMY_LOCATION = "damage.dealt";
+  public static final String WARRIOR_LOCATION = "damage.received";
   private static final Logger LOG = LoggerFactory.getLogger(EnemyVerticle.class);
-  public static final String LOCATION = "damage.done";
-  public static final String LOCATION2 = "damage.received";
+  private static final int MAX_HP = 200;
   private String deploymentId;
-  private final int maxHp;
   private int currentHp;
 
-  public EnemyVerticle() {
-    maxHp = 200;
-    currentHp = maxHp;
+  @Override
+  public void init(Vertx vertx, Context context) {
+    super.init(vertx, context);
+    currentHp = MAX_HP;
+    deploymentId = vertx.getOrCreateContext().deploymentID();
   }
 
   @Override
-  public void start() throws Exception {
-    deploymentId = vertx.getOrCreateContext().deploymentID();
-    consumeMessage();
+  public void start() {
+
+    handlingDamageReceived();
     vertx.setPeriodic(1000, handler -> attack());
     vertx.setPeriodic(5000, handler -> regen());
   }
 
-  public void attack() {
+  private void handlingDamageReceived() {
+    vertx.eventBus().<Integer>consumer(EnemyVerticle.WARRIOR_LOCATION, message -> {
```

Comment:
I think that the `Eventbus` should be an attribute of the class and you should assign it in the `init` method, so you can use it directly instead of calling it from `vertx` instance.